mne/decoding/time_gen.py

```diff
@@ -249,9 +290,25 @@ def predict(self, epochs):
         # Store all testing times parameters
         self.test_times_ = test_times
 
-        # Prepare parallel predictions across time points
+        n_orig_epochs, _, n_times = X.shape
+
+        # Subselects the to-be-predicted epochs so as to manipulate a
+        # contiguous array X by using slices rather than indices.
+        test_epochs_slices = []
+        if self.predict_mode == 'cross-validation':
+            all_test = [ii for train, test in self._splits for ii in test]
+            start = 0
+            for _, test in self._splits:
+                n_test_epochs = len(test)
+                stop = start + n_test_epochs
+                test_epochs_slices.append(slice(start, stop, 1))
+                start += n_test_epochs
+            X = X[all_test]
+
+        # Prepare parallel predictions across testing time points
```

Comment:
The preparation of X to ensure contiguous slicing used to be done within the parallel loop, and was thus redundant and time-costy.